src/DynamoCoreWpf/ViewModels/Preview/WatchViewModel.cs

```diff
@@ -213,6 +237,74 @@ public WatchViewModel(string label, string path, Action<string> tagGeometry, boo
             isCollection = label == WatchViewModel.LIST || label == WatchViewModel.DICTIONARY;
         }
 
+        private static string GetStringFromObject(object obj)
+        {
+            TypeCode type = Type.GetTypeCode(obj.GetType());
+            switch (type)
+            {
+                case TypeCode.Boolean:
+                    return ToString(obj);
+
+                case TypeCode.Double:
+                    return ((double)obj).ToString(numberFormat, CultureInfo.InvariantCulture);
+
+                case TypeCode.Int64:
+                    return ((long)obj).ToString(CultureInfo.InvariantCulture);
+
+                case TypeCode.DateTime:
+                    return ((DateTime)obj).ToString(PreferenceSettings.DefaultDateFormat, CultureInfo.InvariantCulture);
+
+                case TypeCode.Object:
+                    return ToString(obj);
+
+                default:
+                    return (string)obj;
+            };
+        }
+
+        private static string ToString(object obj)
```

Comment:
hmm, I guess this is okay, someone looking quickly at this might think this is overriding the regular ToString